Add automatic redirect creation V2 (#7809)
wagtail/wagtail: docs/reference/contrib/redirects.rst

For an editor's guide to the interface, see :ref:`managing_redirects`.

Automatic redirect creation
===========================

.. versionadded:: 2.16

> By default, Wagtail automatically create permanent redirects for pages (and their descendants) when they are moved or have their slug changed. This helps to preserve SEO rankings of pages over time, and helps site visitors get to the right place when using bookmarks or using outdated links.

Reviewer: ```suggestion
Wagtail automatically creates permanent redirects for pages (and their descendants) when they are moved or their slug is changed. This helps to preserve SEO rankings of pages over time, and helps site visitors get to the right place when using bookmarks or using outdated links.
```

Disabling automatic redirect creation
-------------------------------------

Wagtail's default implementation works best for small-to-medium sized projects (5000 pages or fewer) that mostly use Wagtail's built-in methods for URL generation (Overrides to `Page.get_url_parts()` will be respected, but require additional database queries for each item where custom model fields are used to generate the return value).

If your project does a lot of overriding of URL-related methods, or has sections with tens (or hundreds) of thousands of pages that you want to be able to move freely, you may want to disable this feature by adding the following to your project settings:

.. code-block:: python

>   WAGTAILREDIRECTS_AUTOCREATE = False

Reviewer: Do we need an underscore between `AUTO` and `CREATE`?

Author: Unsure on this one. I was going for something descriptive and compact. I certainly don't mind adding an underscore. Googling around, 'autocreation' appears to be a well-enough known term. 'autocreate' less so, though.
